### PR TITLE
dmq_usrloc: DMQ also copies and sends attributes related to contacts

### PR DESCRIPTION
**Pull request by xkaraman**
#### Pre-Submission Checklist

- [x] Commit message has the format required by CONTRIBUTING guide
- [x] Commits are split per component (core, individual modules, libs, utils, ...)
- [x] Each component has a single commit (if not, squash them into one commit)
- [x] No commits to README files for modules (changes must be done to docbook files
in `doc/` subfolder, the README file is autogenerated)

#### Type Of Change
- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)

#### Checklist:

- [ ] PR should be backported to stable branches
- [x] Tested changes locally
- [x] Related to issue https://github.com/kamailio/kamailio/issues/1968. 

#### Description

This PR fixes the issue in https://github.com/kamailio/kamailio/issues/1968. 

Any feedback is welcome.

### CODE REVIEW

File: src/modules/dmq_usrloc/usrloc_sync.c [hunk 1]
```diff
@@ -825,6 +837,16 @@ int usrloc_dmq_send_contact(
 	srjson_AddNumberToObject(&jdoc, jdoc.root, "reg_id", ptr->reg_id);
 	srjson_AddNumberToObject(&jdoc, jdoc.root, "server_id", ptr->server_id);
 
+	/*  Loop through Χavp attributes of the contact and and create a json object */
+	srjson_t *jdoc_xavp = srjson_CreateObject(&jdoc);
```
Comment (miconda) [inline src/modules/dmq_usrloc/usrloc_sync.c:841]: Variable should be declared at the beginning of the block to be coherent and also keep compliant with older C standards.

**Comment by miconda** [comment]
The commit for usrloc #3d79568 does not only change the destroy order, is it also fixing something?

File: src/modules/dmq_usrloc/usrloc_sync.c (hunk 1 above)
Comment (xkaraman) [reply src/modules/dmq_usrloc/usrloc_sync.c]: Fixed on some other variables too on the newly static functions.

**Comment by xkaraman** [comment]
Hey Daniel,
 
Besides the order change of destroy, it copies to the new contact the `xavp` from incoming `ucontact_t` argument just like the rest of the fields. 

If the config variable `xavp_contact` is found and not empty, then `ucontact_xavp_store` destroy this xavp and fills it up with whatever is contained inside. 

Some doubts i had, is whether i shall `c->xavp = _ci->xavp;` or `	_c->xavp = xavp_clone_level_nodata(_ci->xavp);` or `xavp_add(_ci->xavp, &c->xavp)`. What is the difference between all these methods?

**Comment by henningw** [comment]
Are there more comments or objections? If not I would merge it in the next days.

**Comment by miconda** [comment]
I haven't gotten the time to review if the list of XAVPs can be just linked or needs to be cloned. If it is just used temporarily in a function, then linking should be fine, but if it needs to be kept for long, then it could be that clone has to be done, otherwise the list stays linked from two places and attempted to be destroyed twice.

If the above is clarified, I am fine to merge it.

**Comment by henningw** [comment]
Thanks, yes this was basically the question also @xkaraman had earlier. The XAVPs will be used in the cfg script context later on I think. He will do another review.

**Comment by xkaraman** [comment]
Hi @miconda,

What does long mean in this case? What does `xavp_clone_level_nodata` achieve in this regard?

Thanks!

**Comment by henningw** [comment]
The PR was reviewed and discussed together with Xenofon, some comments have been added and it was pushed in the referenced commits. Close this PR.

**Pull request closed without merging**